airflow/datasets/__init__.py

```diff
@@ -294,6 +297,9 @@ def as_expression(self) -> Any:
     def iter_datasets(self) -> Iterator[tuple[str, Dataset]]:
         yield self.uri, self
 
+    def iter_dataset_aliases(self) -> Iterator[DatasetAlias]:
+        raise StopIteration()
```

Comment:
```suggestion
    def iter_dataset_aliases(self) -> Iterator[DatasetAlias]:
        return iter(())
```

I think? Otherwise the error is raised to early.

```pycon
>>> def f1():
...     raise StopIteration
...     
>>> def f2():
...     return iter(())
...     
>>> for a in f1(): pass  # Wrong
... 
Traceback (most recent call last):
  File "<python-input-2>", line 1, in <module>
    for a in f1(): pass
             ~~^^
  File "<python-input-0>", line 2, in f1
    raise StopIteration
StopIteration
>>> for a in f2(): pass  # Correct
... 
>>> 
```

Reply:
Alternatively I think this would work

```python
def iter_....:
    return
    yield  # Hack to make Python turn this into a generator function.
```

Reply:
yep, thanks for pointing it out! I might use the wrong example to test this out 🤦‍♂️ Just fixed it!